docs/examples/sharing-state-between-islands.md

```diff
@@ -0,0 +1,186 @@
+---
+description: |
+  When you need to have state shared between islands, this page provides a few recipes.
+---
+
+All of this content is lifted from this great
+[example](https://fresh-with-signals.deno.dev/) by Luca. The source can be found
+[here](https://github.com/lucacasonato/fresh-with-signals).
+
+## Multiple Sibling Islands with Independent State
+
+Imagine we have `Counter.tsx` like this:
+
+```ts
+import { useSignal } from "@preact/signals";
+import { Button } from "../components/Button.tsx";
+
+interface CounterProps {
+  start: number;
+}
+
+// This island is used to display a counter and increment/decrement it. The
+// state for the counter is stored locally in this island.
+export default function Counter(props: CounterProps) {
+  const count = useSignal(props.start);
+  return (
+    <div class="flex gap-2 items-center w-full">
+      <p class="flex-grow-1 font-bold text-xl">{count}</p>
+      <Button onClick={() => count.value--}>-1</Button>
+      <Button onClick={() => count.value++}>+1</Button>
+    </div>
+  );
+}
+```
+
+Note how `useSignal` is within the `Counter` component. Then if we instantiate
+some counters like this...
+
+```ts
+<Counter start={3} />
+<Counter start={4} />
+```
+
+they'll keep track of their own independent state. Not much sharing going on
+here, yet.
+
+## Multiple Sibling Islands with Shared State
+
+But we can switch things up by looking at a `Slider.tsx` like this:
+
+```ts
+import { signal } from "@preact/signals";
+
+const slider = signal(50);
```

Comment:
Whilst this is a valid pattern, let's push folks to use `useSignal` instead. Like a good example would be to instantiate `useSignal` inside a route file and then pass that to two different islands.

The signal here is not just shared between islands, but also for every request, which is likely unintended.

Reply:
Sadly I don't follow. This is in an island. How is the signal shared across every request?

I can rework this to use `useSignal` and then pass it in instead.

Reply:
Right, I missed that this is in an island. Still showing how signals are shared outside of that is more useful which means they'll be passed as props to the island.